app/src/test/java/org/pesmypetcare/mypetcare/services/StubCommunityService.java

```diff
@@ -319,6 +325,102 @@ public void reportPost(User user, Post post, String reportMessage) {
+        boolean found = false;
+
+        for (Forum forum : groups.get(groupIndex).getForums()) {
+            if (forum.getName().equals(post.getForum().getName())) {
+                for (Post forumPost : forum.getPosts()) {
+                    if (forumPost.getUsername().equals(post.getUsername())
+                    && forumPost.getCreationDate().compareTo(post.getCreationDate()) == 0) {
+                        post.setPostImage(image);
+                        found = true;
+                        break;
+                    }
+                }
+            }
+        }
+        if (!found) {
+            throw new PostNotFoundException();
+        }
+    }
+
+    @Override
+    public void deletePostImage(User user, Post post) throws PostNotFoundException {
+        int groupIndex = groups.indexOf(post.getForum().getGroup());
+        boolean found = false;
+
+        for (Forum forum : groups.get(groupIndex).getForums()) {
+            if (forum.getName().equals(post.getForum().getName())) {
+                for (Post forumPost : forum.getPosts()) {
+                    if (forumPost.getUsername().equals(post.getUsername())
+                        && forumPost.getCreationDate().compareTo(post.getCreationDate()) == 0) {
+                        post.setPostImage(null);
+                        found = true;
+                        break;
+                    }
+                }
+            }
+        }
+        if (!found) {
+            throw new PostNotFoundException();
+        }
+    }
+
+    @Override
+    public void addGroupImage(User user, Group group, Bitmap image) throws GroupNotFoundException {
+        int groupIndex = groups.indexOf(group);
+        if (groupIndex == -1) {
+            throw new GroupNotFoundException();
+        } else {
+            groups.get(groupIndex).setGroupIcon(image);
+        }
+    }
+
+    @Override
+    public void deleteGroupImage(User user, Group group) throws GroupNotFoundException {
+        int groupIndex = groups.indexOf(group);
+        if (groupIndex == -1) {
+            throw new GroupNotFoundException();
+        } else {
+            groups.get(groupIndex).setGroupIcon(null);
+        }
+    }
+
+    @Override
+    public byte[] getPostImage(User user, Post post, MessageDisplay messageDisplay) {
+        int groupIndex = groups.indexOf(post.getForum().getGroup());
+        Forum selectedForum = null;
+
+        for (Forum forum : groups.get(groupIndex).getForums()) {
+            if (post.getForum().getName().equals(forum.getName())) {
+                selectedForum = forum;
+                break;
+            }
+        }
+
+        for (Post forumPost : Objects.requireNonNull(selectedForum).getPosts()) {
+            if (forumPost.getCreationDate().compareTo(post.getCreationDate()) == 0) {
+                return new byte[] {0x00};
+            }
+        }
+
+        return null;
+    }
+
+    @Override
+    public byte[] getGroupImage(User user, Group group) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Return an empty array rather than null.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5529921)